src/gluonts/support/util.py

```diff
@@ -177,239 +177,245 @@ def erfinv(x: np.array) -> np.array:
-                self.quantiles, self.quantiles[1:]
-            ):
-                if (
-                    current_quantile <= inference_quantile
-                    and inference_quantile < next_quantile
-                ):
-                    weights = [
-                        (next_quantile - inference_quantile)
-                        / (next_quantile - current_quantile),
-                        (inference_quantile - current_quantile)
-                        / (next_quantile - current_quantile),
-                    ]
-                    return (
-                        weights[0]
-                        * self.quantile_predictions[str(current_quantile)]
-                        + weights[1]
-                        * self.quantile_predictions[str(next_quantile)]
-                    )
+            for i, (x1, x2) in enumerate(zip(self.x_coord, self.x_coord[1:])):
+                if x1 < x < x2:
+                    denominator = x2 - x1 + self.tol
+                    return (x2 - x) / denominator * self.y_coord[i] + (
+                        x - x1
+                    ) / denominator * self.y_coord[i + 1]
 
 
 class TailApproximation:
     """
-    Apporoximate quantile function on tails based on trained quantile predictions
-    and make a inference on query point.
+    Approximate function on tails based on knots and make a inference on query point.
     Can be used for either interpolation or extrapolation on tails
 
     Parameters
     ----------
-    quantile_predictions
-        Dict with key = str(quantile) and value is np.array with size prediction_length
+    x_coord
+        x-coordinates of the data points must be in increasing order.
+    y_coord
+        y-coordinates of the data points - may be a higher numpy array.
     approximation_type
         str of tail approximation type
 
     """
 
     def __init__(
         self,
-        quantile_predictions: dict,
+        x_coord: List[float],
+        y_coord: List[np.ndarray],
         approximation_type: str = "exponential",
         tol: float = 1e-8,
-    ):
-        self.quantile_predictions = quantile_predictions
-        self.quantiles = sorted(map(float, self.quantile_predictions))
-        self.num_quantiles = len(self.quantile_predictions)
+    ) -> None:
+        self.x_coord = x_coord
+        assert sorted(self.x_coord) == self.x_coord
+        self.y_coord = y_coord
+        self.num_points = len(self.x_coord)
         self.approximation_type = (
-            approximation_type if self.num_quantiles > 1 else None
+            approximation_type if len(self.x_coord) > 1 else None
         )
         self.tol = tol
         if not self.approximation_type:
             pass
         elif self.approximation_type == "exponential":
-            self.init_exponential_tail_weights()
             self.tail_function_left = self.exponential_tail_left
             self.tail_function_right = self.exponential_tail_right
```

Comment:
Similarly to the interpolation class: I think we could avoid putting too much logic and options in this class. Other tail approximations can be added as new classes; they can share the interface, and if some code is also to be shared among them, we will figure out how to better factor things.

So I think this class could be called `ExponentialTailApproximation`, right?

Reply:
yes will clean it up and update - thanks!